Implemented divide method for CalcLength class

njlawton opened the pull request:
This method is similar in implementation to multiply method

Commit:
Created shell for CalcLength divide method
Cretaed basic shell for method and added basic unit tests to test its functionailty

Commit:
Fixed test cases for CalcLength division

Commit:
Implemented division method for CalcLength class
Implementation has been implemented and has passed its unit tests

Commit:
Fixed test case variable to start with a lower case letter

Commit:
Changed layout of devide method in class ClacLength
Changed method layout including loop type and way of constructing the calcdictionary object to be consistant with the multiply method in CalcLength

wilddamon commented on test/js/calc-length.js:
```diff
@@ -66,11 +66,21 @@ suite('CalcLength', function() {
   });
 
-  test('Multiplication of a decimal number produses expected result', function() {
+  test('Multiplication by a decimal number produses expected result', function() {
```
s/produses/produces again, I guess we missed one :p

wilddamon commented on test/js/calc-length.js:
```diff
@@ -66,11 +66,21 @@ suite('CalcLength', function() {
+    assert.strictEqual(dividedCLenght.cssString, 'calc(2.5px+1em)');
+  });
 
+  test('Division by a decimal number produses expected result', function() {
```
here too

wilddamon commented on test/js/calc-length.js:
```diff
@@ -66,11 +66,21 @@ suite('CalcLength', function() {
-  test('Multiplication of a decimal number produses expected result', function() {
+  test('Multiplication by a decimal number produses expected result', function() {
     var cLength = new CalcLength({px: 10, em: 3.2})
     var multipliedCLenght = cLength.multiply(0.5);
```
Unfortunately the style is not to contract variable names, so these should be

calcLength
multipliedCalcLength

or some other name that uses full words. Do you mind fixing up the other places in this file while you're here?

Thanks!

rjwright replied:
There's also a typo here

```
multipliedCLenght
```

Commit:
Fixed spelling mistakes in test cases for the CalcLength class

Commit:
Changed some variable names in test classes to conform with code style
Many variable names used short versions of words were as the code standard used requires variables use unshortened words

wilddamon commented on test/js/calc-length.js:
```diff
@@ -1,9 +1,9 @@
-    assert.instanceOf(calcLen, CalcLength, 'A new CalcLength should be an instance of CalcLength');
-    assert.instanceOf(calcLen, LengthValue, 'A new CalcLength should be an instance of LengthValue');
-    assert.instanceOf(calcLen, StyleValue, 'A new CalcLength should be an instance of StyleValue');
+    var calcLenght = new CalcLength({px: 10});
```
s/calcLengt/calcLength - missing the final "h". Same elsewhere

wilddamon commented on test/js/calc-length.js:
```diff
@@ -13,12 +13,12 @@ suite('CalcLength', function() {
   test('CalcLength empty constructor initializes other fields to null', function() {
-    var lenFromEmptyDict;
-    assert.doesNotThrow(function() {lenFromEmptyDict = new CalcLength({px: 10})});
+    var lengthFromEmptyDictionary;
```
The dictionary isn't empty. How about just calling this calcLength as well? Possibly the name of the test could be changed too - how about just

"CalcLength constructor initializes other fields to null"

wilddamon commented on test/js/calc-length.js:
```diff
@@ -61,16 +61,26 @@ suite('CalcLength', function() {
-    var cLength = new CalcLength({px: 10, em: 3.2})
-    var multipliedCLenght = cLength.multiply(4);
+    var calcLength = new CalcLength({px: 10, em: 3.2})
+    var multipliedCLenght = calcLength.multiply(4);
```
multipliedCalcLength

wilddamon commented on test/js/calc-length.js:
```diff
@@ -61,16 +61,26 @@ suite('CalcLength', function() {
     assert.strictEqual(multipliedCLenght.cssString, 'calc(40px+12.8em)');
   });
 
   test('Multiplication of a decimal number produses expected result', function() {
```
s/produses/produces

wilddamon commented on test/js/calc-length.js:
```diff
@@ -61,16 +61,26 @@ suite('CalcLength', function() {
-    var cLength = new CalcLength({px: 10, em: 3.2})
-    var multipliedCLenght = cLength.multiply(0.5);
+    var calcLength = new CalcLength({px: 10, em: 3.2})
+    var multipliedCLenght = calcLength.multiply(0.5);
```
multipliedCalcLength

wilddamon commented on test/js/calc-length.js:
```diff
@@ -61,16 +61,26 @@ suite('CalcLength', function() {
 
+  test('Division of a calc-length length produces a new calc-length object', function() {
+    var calcLength = new CalcLength({px: 10, em: 4.0})
+    var dividedCLenght = calcLength.divide(4);
```
dividedCalcLength

wilddamon commented on test/js/calc-length.js:
```diff
@@ -61,16 +61,26 @@ suite('CalcLength', function() {
+    assert.strictEqual(dividedCLenght.cssString, 'calc(2.5px+1em)');
+  });
 
+  test('Division of a decimal number produses expected result', function() {
```
s/produses/produces

wilddamon commented on test/js/length-value.js:
```diff
@@ -1,15 +1,15 @@
-    assert.instanceOf(fromVal, SimpleLength, 'A new fromValue should be an instance of SimpleLength');
-    assert.instanceOf(fromVal, LengthValue, 'A new fromValue should be an instance of LengthValue');
-    assert.instanceOf(fromVal, StyleValue, 'A new fromValue should be an instance of StyleValue');
+    var fromValue = (new LengthValue()).fromValue(9.2, 'px');
```
should be fine to just call this value, here and elsewhere. "fromDictionary" sounds like it should be a dictionary, which is a bit confusing...

Commit:
Made small improvements in test cases to improve understandability
Changed some variable names to better represent there function and to adhere to code style guides, Fixed some spelling mistakes

Commit:
Added more small improvements to test suite
Small changes to grammer and spelling have been made to improve readability

Commit:
Fixed broken test case in length-value.js
Last commit accedentaly changed a method witch broke a test in length-value.jsthis commit fixes this issue

wilddamon commented on src/calc-length.js:
```diff
@@ -79,6 +79,20 @@
+  CalcLength.prototype.divide = function(divider) {
+    var calcDictionary = {};
+
+     //iterate through all length types and divide all non null lengths 
```
Nitpick: Comment should start with capital, should have one space between the // and the text

wilddamon commented on test/js/calc-length.js:
```diff
@@ -60,17 +60,27 @@ suite('CalcLength', function() {
 
+  test('Division of a CalcLength length produces a new CalcLength object', function() {
+    var calcLength = new CalcLength({px: 10, em: 4.0})
+    var dividedCalcLenght = calcLength.divide(4);
```
Typo: s/dividedCalcLenght/dividedCalcLength

t and h around the wrong way.

wilddamon commented on test/js/length-value.js:
```diff
@@ -1,15 +1,15 @@
-    assert.instanceOf(fromVal, LengthValue, 'A new fromValue should be an instance of LengthValue');
-    assert.instanceOf(fromVal, StyleValue, 'A new fromValue should be an instance of StyleValue');
+  test('fromDictionary returns instanceof a LengthValue and StyleValue', function() {
+    var fromDictionary = (new LengthValue()).fromValue(9.2, 'px');
```
Previous comment not addressed: Please don't call this "fromDictionary" - it sounds like the variable will be a dictionary, which it isn't. Please call this "simpleLength" or just "value"

wilddamon replied:
Also, you should be able to just write

LengthValue.fromValue(9.2, 'px')

instead of also creating a LengthValue then calling the static method on it.

wilddamon commented on test/js/length-value.js:
```diff
@@ -1,15 +1,15 @@
-    assert.instanceOf(fromDict, CalcLength, 'A new fromDictionary should be an instance of CalcLength');
-    assert.instanceOf(fromDict, LengthValue, 'A new fromDictionary should be an instance of LengthValue');
-    assert.instanceOf(fromDict, StyleValue, 'A new fromDictionary should be an instance of StyleValue');
+    var fromDictionary = (new LengthValue()).fromDictionary({px: 10});
```
Both above comments also apply here, except you can call this on calcLength or value.

Commit:
Small improvements to commenting layout in CalcLength class

Commit:
More minor improvements to the test suite
Some variable names in length-value.js were changed due to a mistaken name change in a previous commit, Some testing explanations were also updated to better reflect the function of that test

wilddamon commented:
LGTM
